airbyte-integrations/connectors/source-doordash/source_doordash/source.py

```diff
@@ -0,0 +1,171 @@
+        # on the destination side will handle all the clean up.
+        temp_date = datetime.strptime(start_date, "%Y-%m-%d")
+        temp_date = temp_date - timedelta(days=7)
+        start_date = temp_date.astimezone(pytz.utc).strftime("%Y-%m-%d")
+
+        end_date = datetime.now().astimezone(pytz.utc).strftime("%Y-%m-%d")
+
+        request_json = { 
+            "store_ids": store_ids,
+            "start_date": start_date,
+            "end_date": end_date,
+            "report_type": self.report_type
+        }
+        self.logger.info(f"Sending request: {request_json}")
+        return request_json
+    
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None    
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        # Initial create report request would provide a report ID.
+        report_id = response.json()['report_id']
+        self.logger.info(f"Create report request completed. Report ID: {report_id}.")
+        get_report_url = f"{self.url_base}/{report_id}/reportlink"
+
+        # Check to see if the report is ready for download with an exponential backoff. Loop until it's ready. Also 
+        # provide a way out of the loop if it takes more than 10 minutes.
+        delay = 1
+        time_start = time.time()
+        time_current = time_start
+        get_report_response = requests.get(get_report_url, headers=self.authenticator.get_auth_header())
+        while not get_report_response.ok or (get_report_response.json()['report_status'] != 'SUCCEEDED' and time_current - time_start < 600): 
+            time.sleep(delay)
+            delay += delay
+            get_report_response = requests.get(get_report_url, headers=self.authenticator.get_auth_header())
+            time_current = time.time()
+
+        # Terminate further processing if we exit the loop due to timeout.
+        if get_report_response.json()['report_status'] != 'SUCCEEDED':
+            msg = 'DoorDash does not provide a report download URL in 10 minutes. Terminating.'
+            self.logger.error(msg)
+            raise Exception(msg)
+
+        # Download the report as a temp file.        
+        report_link = get_report_response.json()['report_link']
+        self.logger.info(f'Report ready for download. URL: {report_link}.')
+        report_download_response = requests.get(report_link, headers=self.authenticator.get_auth_header())
+        with tempfile.TemporaryDirectory() as temp_dir:
+            with tempfile.TemporaryFile(dir=temp_dir, suffix='.zip') as report_zip_file:
+                report_zip_file.write(report_download_response.content)
+                report_zip_file.flush()
+                self.logger.info(f'Report download. Zip file written to {report_zip_file.name}.')
+
+                # Unzip the zip file.
+                with zipfile.ZipFile(report_zip_file, 'r') as zip_obj:
```

Comment:
What are the performance implications of a lot of this unarchiving?

Reply:
Yeah good points. The whole thing is based on how DoorDash's API works, so we don't really have any other options. Fortunately, the export is rather small (60MB). They hold a max of 2 years of data. With us starting in 2022, I think we will probably max out at 200MB.

It comes across as a single zip file that would unzip into a single CSV.

I don't know if there's any other efficient way to unzip and read the CSV in Python, so please let me know if you have any suggestions.

Reply:
@tayeke Would pandas be a potential performance improvement here?